tools/migrate_branch.py

```diff
@@ -0,0 +1,265 @@
+        self.gh_source_branch = source
+        self.gh_target_branch = target
+        self.gh_org = target_org or 'OCA'
+
+    def _replace_content(self, repo, path, replace_list):
+        # Re-read path for retrieving content
+        gh_file = repo.contents(path, self.gh_target_branch)
+        content = gh_file.decoded
+        for replace in replace_list:
+            content = re.sub(replace[0], replace[1], content, flags=re.DOTALL)
+        new_file_blob = repo.create_blob(content, encoding='utf-8')
+        return {
+            'path': path,
+            'mode': '100644',
+            'type': 'blob',
+            'sha': new_file_blob
+        }
+
+    def _create_commit(self, repo, tree_data, message):
+        if tree_data:
+            branch = repo.branch(self.gh_target_branch)
+            tree_sha = branch.commit.commit.tree.sha
+            tree = repo.create_tree(tree_data, tree_sha)
+            commit = repo.create_commit(
+                message=message, tree=tree.sha, parents=[branch.commit.sha],
+                author=self.gh_credentials, committer=self.gh_credentials)
+            repo.ref('heads/{}'.format(branch.name)).update(commit.sha)
+            return commit
+
+    def _mark_modules_uninstallable(self, repo, root_contents):
+        """Make uninstallable the existing modules in the repo."""
+        tree_data = []
+        modules = []
+        for root_content in root_contents.values():
+            if root_content.type != 'dir':
+                continue
+            module_contents = repo.contents(
+                root_content.path, self.gh_target_branch)
+            manifest = module_contents.get('__openerp__.py')
+            if manifest:
+                modules.append(root_content.path)
+                # Re-read path for retrieving content
+                tree_data.append(self._replace_content(
+                    repo, manifest.path,
+                    [('["\"]installable["\"]:[ ]*True',
+                      "'installable': False")]))
+        self._create_commit(
+            repo, tree_data, "[MIG] Make modules uninstallable")
+        return modules
+
+    def _update_metafiles(self, repo, root_contents):
+        """Update metafiles (README.md, .travis.yml...) for pointing to
+        the new branch.
+        """
+        tree_data = []
+        source_string = self.gh_source_branch.replace('.', '\.')
+        target_string = self.gh_target_branch
+        source_string_dash = self.gh_source_branch.replace('.', '-')
+        target_string_dash = self.gh_target_branch.replace('.', '-')
+        if root_contents.get('README.md'):
+            tree_data.append(self._replace_content(
+                repo, 'README.md',
+                [(source_string, target_string),
+                 (source_string_dash, target_string_dash),
+                 ("\[//]: # \(addons\).*\[//]: # \(end addons\)",
+                  "[//]: # (addons)\n[//]: # (end addons)")]))
```

Comment:
Instead of doing this replacement, run oca-gen-addons-table on the new branch and after setting installable=False in the manifests.